Formula/openfast.rb

```diff
@@ -0,0 +1,32 @@
+class Openfast < Formula
+  desc "NREL-supported OpenFAST whole-turbine simulation code."
+  homepage "http://openfast.readthedocs.io"
+  url "https://github.com/openfast/openfast.git",
+    :using => :git,
+    :branch => "dev"
+    # :tag => "v2.2.0"
+  # sha256 "9debbe44643995add15b0651bb64c1b6f68f3eb29a8ff650e7e7f9fff8c8c990"
+  version "v2.3.0"
+
+  depends_on "cmake" => :build
+  depends_on "gcc"
+  depends_on "openblas"
+  depends_on :xcode
+
+  def install
+    args = std_cmake_args + %W[
+      -DCMAKE_BUILD_TYPE=RelWithDebInfo
+      -DBLA_VENDOR=OpenBLAS
+    ]
+
+    mkdir "build" do
+      system "cmake", "..", *args
+      system "make install"
+    end
+  end
+  
+  test do
+    system bin/"openfast", "-h"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
The formula cookbook says that "a bad test is better than no test at all," but your point is taken.

I've created a very simple test case, but it still requires four input files. Whats a good way to get them? Should they be created by the formula or downloaded from a repository? The test can check that the output file is created, as you've suggested.

Reply:
> The formula cookbook says that "a bad test is better than no test at all,"

That's why some old formula use version or help tests. We try to do better these days :smile: 

Either creating or downloading is fine.

Reply:
:) You're right, a bad test is still bad. I found this [example formula](https://github.com/syhw/homebrew/blob/master/Library/Contributions/example-formula.rb) and made an attempt at generating the test case files.

Reply:
That's a lot of test info 😨, looks like a good test though.

Reply:
Ya, not ideal, but its a simple and fully functional test case. Is there a way to define the input file text as a variable outside of the class block? It would be nice to move all that text out of the way.

Reply:
If it's also available somewhere else you could use a `resource` block to pull it in. Making it a variable won't change much otherwise IYAM